app/models/landing_page/block/hero.rb

```diff
@@ -3,21 +3,27 @@ module LandingPage::Block
   HeroImage = Data.define(:alt, :sources)
 
   class Hero < Base
-    attr_reader :image, :hero_content, :theme, :theme_colour
+    attr_reader :image, :theme, :theme_colour
 
     def initialize(block_hash, landing_page)
       super
 
       alt, sources = data.fetch("image").values_at("alt", "sources")
       sources = HeroImageSources.new(**sources)
       @image = HeroImage.new(alt:, sources:)
-      @hero_content = LandingPage::BlockFactory.build_all(data.dig("hero_content", "blocks"), landing_page)
```

Comment:
If you wanted to still do this in the constructor / with an attr_reader, you could use my new favourite ruby idiom:

```
@hero_content = data.dig("hero_content", "blocks")&.then do |blocks|
    LandingPage::BlockFactory.build_all(blocks, landing_page)
  end
```

(The & means it will be nil if the result of dig is nil, and the .then means it will be the result of the block if dig returns something other than nil)

Reply:
Although I guess this doesn't cover the case where `.dig("hero_content", "blocks")` returns `[]`.

If you wanted to do that you'd have to do:

```
@hero_content = data.dig("hero_content", "blocks").then do |blocks|
    return if blocks.blank?
    
    LandingPage::BlockFactory.build_all(blocks, landing_page)
  end
```

(Basically like you have in your method below)

Reply:
I'm not a big fan of multi-line assignments in the constructor. I'd put it in a method.